srt-tokio/src/builder.rs

```diff
@@ -58,11 +58,20 @@ use srt_protocol::{
 #[derive(Debug, Clone)]
 #[must_use]
 pub struct SrtSocketBuilder {
-    local_addr: SocketAddr,
+    local_addr: Option<IpAddr>,
+    local_port: u16,
     conn_type: ConnInitMethod,
     init_settings: ConnInitSettings,
 }
 
+fn unspecified(orig: IpAddr) -> IpAddr {
```

Comment:
Would be cleaner if this just took a `is_ipv4` or `is_ipv6` bool, unless there's a compelling reason that you chose this